subprojects/gradle/buildchecks/src/main/kotlin/com/avito/android/plugin/build_param_check/OsxLocalhostResolvingTask.kt

```diff
@@ -0,0 +1,53 @@
+package com.avito.android.plugin.build_param_check
+
+import com.avito.android.plugin.build_param_check.OsxLocalhostResolvingTask.Action.Parameters
+import org.gradle.api.DefaultTask
+import org.gradle.api.GradleException
+import org.gradle.api.tasks.TaskAction
+import org.gradle.workers.WorkAction
+import org.gradle.workers.WorkParameters
+import org.gradle.workers.WorkerExecutor
+import java.net.InetAddress
+import javax.inject.Inject
+import kotlin.system.measureTimeMillis
+
+abstract class OsxLocalhostResolvingTask @Inject constructor(
+    private val workerExecutor: WorkerExecutor
+) : DefaultTask() {
+
+    @TaskAction
+    fun check() {
+        @Suppress("UnstableApiUsage")
+        workerExecutor.noIsolation().submit(Action::class.java) {}
+    }
+
+    @Suppress("UnstableApiUsage")
+    abstract class Action : WorkAction<Parameters> {
```

Comment:
Boilerplate signals that we need helper classes to work with workers

Reply:
Agree, I'll create an issue